include/libtorrent/torrent.hpp

```diff
@@ -1351,10 +1351,10 @@ namespace libtorrent
 		file_index_t m_error_file;
 
 		// the average time it takes to download one time critical piece
-		std::uint32_t m_average_piece_time = 0;
+		std::int32_t m_average_piece_time = 0;
```

Comment:
hm.. this should really be a ``std::chrono::seconds``. but I guess that's for a separate patch. this is a step in the right direction though, since those are signed